pkgs/development/interpreters/python/hooks/wheel-deps-versions-hook.sh

```diff
@@ -0,0 +1,82 @@
+# shellcheck shell=bash
+
+# Setup hook that modifies Python dependencies versions.
+#
+# Example usage in a derivation:
+#
+#   { …, pythonPackages, … }:
+#
+#   pythonPackages.buildPythonPackage {
+#     …
+#     nativeBuildInputs = [ pythonPackages.wheelDepsVersionsHook ];
+#
+#     # This will relax the dependency restrictions
+#     # e.g.: abc>1,<=2 -> abc
+#     relaxPythonDeps = [ "abc" ];
+#     # This will relax all dependencies restrictions instead
+#     # relaxPythonDeps = true;
+#     # This will remove the dependency
+#     # e.g.: cde>1,<=2 -> <nothing>
+#     removePythonDeps = [ "cde" ];
+#     # This will remove all dependencies from the project
+#     # removePythonDeps = true;
+#     …
+#   }
+
+wheelRelaxDepsVersions() {
+    local -r metadata_file="$1"
+
+    if [[ -z "$relaxPythonDeps" ]] || [[ "$relaxPythonDeps" == 0 ]]; then
+        return
+    elif [[ "$relaxPythonDeps" == 1 ]]; then
+        sed -i "$metadata_file" -r \
+            -e 's/(Requires-Dist: \S*) \(.*\)/\1/'
+    else
+        for dep in $relaxPythonDeps; do
+            sed -i "$metadata_file" -r \
+                -e "s/(Requires-Dist: $dep) \(.*\)/\1/"
+        done
+    fi
+}
+
+wheelRemoveDepsVersions() {
+    local -r metadata_file="$1"
+
+    if [[ -z "$removePythonDeps" ]] || [[ "$removePythonDeps" == 0 ]]; then
+        return
+    elif [[ "$removePythonDeps" == 1 ]]; then
+        sed -i "$metadata_file" \
+            -e '/Requires-Dist:.*/d'
+    else
+        for dep in $removePythonDeps; do
+            sed -i "$metadata_file" \
+                -e "/Requires-Dist: $dep/d"
+        done
+    fi
+
+}
+
+wheelDepsVersionsHook() {
+    pushd dist
+
+    local -r package="$pname-$version"
```

Comment:
Do we still need to fallback to `name` attribute?

The reason I am not using the `name` attribute directly here is because in `pythonPackages` the name attribute will be prefixed with the interpreter (e.g. `python3Packages.abc` will be `python-3.9-abc`). However top-level Python applications have their `name` attribute correctly (e.g.: without interpreter prefix).

I don't think this will be an issue because this will only be used by either new or maintained packages, so in the rare case that we still have some package having only the `name` attribute this would be a good reason to migrate to the new format.